src/sentry/types/region.py

```diff
@@ -153,21 +150,25 @@ def get_region_by_name(name: str) -> Region:
         raise RegionResolutionError(f"No region with name: {name!r}")
 
 
-def get_region_for_organization(organization: Organization) -> Region:
+@control_silo_function
+def get_region_for_organization(organization_slug: str) -> Region:
+    from sentry.models.organizationmapping import OrganizationMapping
+
     """Resolve an organization to the region where its data is stored.
 
     Raises RegionContextError if this Sentry platform instance is configured to
     run only in monolith mode.
     """
-    mapping = load_global_regions()
+    directory = load_global_regions()
 
-    if not mapping.regions:
+    if not directory.regions:
         raise RegionContextError("No regions are configured")
 
-    # Backend representation to be determined. If you are working on code
-    # that depends on this method, you can mock it out in unit tests or
```

Comment:
Did you audit for existing mocks? If there are any, we should consider ways to swap out an in-test alternative implementation of this that actually invokes the endpoints w/o http

Reply:
@mikejihbe Replaced the existing mocks so they use this method. I also chose slug only because it's what the API Gateway uses to proxy requests (since the org slug is in the path).